resolver/resolver.go

```diff
@@ -378,7 +378,8 @@ func (res *Resolver) handleEmpty(rs *records.RecordGenerator, name string, m, r
 	}
 
 	m.Rcode = dns.RcodeNameError
-	if qType == dns.TypeAAAA && len(rs.SRVs[name])+len(rs.As[name]) > 0 {
+
+	if (qType == dns.TypeAAAA) || (len(rs.SRVs[name])+len(rs.As[name]) > 0) {
```

Comment:
:+1: 
probably worth a comment explaining why this behavior is the way that it is. including a link to the relevant RFC